include/hipSYCL/sycl/libkernel/host/builtins.hpp

```diff
@@ -612,6 +612,15 @@ HIPSYCL_BUILTIN T __hipsycl_mul24(T x, T y) noexcept {
   return x * y;
 }
 
+template<class T, std::enable_if_t<std::is_integral_v<T>,int> = 0>
+HIPSYCL_BUILTIN T __hipsycl_popcount(T x) noexcept {
+    T count = T(0);
+    for (size_t i = 0; i < sizeof(T) * 8; i++) {
+        count += !!(x & (T(1) << i));
```

Comment:
What is the purpose of the two `!!` ?

Reply:
Convert any non-0 value to 1 and 0 to 0.

Reply:
Ah, it's needed to clamp values to `[0,1]` because `x & mask` may result in a large number. I get it now, thanks!